lib/usagereporter/usagereporter.go

```diff
@@ -243,6 +258,10 @@ func (r *UsageReporter[T]) Run(ctx context.Context) {
 		select {
 		case <-ctx.Done():
 			return
+		case <-r.gracefulStopScheduled:
+			r.enqueueBatch()
+			r.gracefulStopEnqueued <- true
```

Comment:
Does this ordering guarantee that we hit the `<-r.submissionQueue` branch in the `runSubmit` select?

Reply:
I got rid of this channel, now the context of `runSubmit` goroutine is cancelled, but the idea is the same.
First enqueue a batch and then stop ` runSubmit` right after the submit finishes. Do you see any issues with ordering?